nx/lib/nx/serving.ex

```diff
@@ -562,7 +686,57 @@ defmodule Nx.Serving do
       {:DOWN, ^ref, _, _, reason} ->
         # We fake monitor messages, so still demonitor and flush.
         Process.demonitor(ref, [:flush])
-        exit({reason, {__MODULE__, :batched_run, [name, input]}})
+        {:DOWN, reason}
+    end
+  end
+
+  defp distributed_batched_run!(name, input, distributed_callback) do
+    distributed_batched_run_with_retries!(name, distributed_callback.(input), 3)
+  end
+
+  defp distributed_batched_run_with_retries!(name, input, 0) do
+    exit({:noproc, {__MODULE__, :distributed_batched_run, [name, input, [retries: 0]]}})
+  end
+
+  defp distributed_batched_run_with_retries!(name, input, retries) do
+    case :pg.get_members(Nx.PG, __MODULE__) do
+      [] ->
+        exit({:noproc, {__MODULE__, :distributed_batched_run, [name, input, [retries: retries]]}})
+
+      entries ->
+        pid = Enum.fetch!(entries, :rand.uniform(length(entries)) - 1)
```

Comment:
```suggestion
        pid = Enum.random(entries)
```

Reply:
My version is a bit more optimized because we know the list is small. The Enum version assumes we don't know the list size.

Reply:
Both `Enum.fetch!` and `Enum.random` do `drop_list` with a random number no?

Reply:
Gah, good call. I missed the special case for lists in there. :)